scripts/core/sharedContext.yml

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension for shared contexts."
+version: "1.6"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Shared Contexts Extension Name"
+version: "1.6"
+name: $X_SHARED_CONTEXTS_EXP_NAME
+value: '"$X_experimental_shared_contexts"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Shared Contexts Extension Version(s)"
+version: "1.6"
+name: $x_shared_contexts_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Defines a shared context for a set of context handles."
+version: "1.6"
+class: $xDriver
+name: GetSharedContextExp
+details:
+    -  |
+       If a shared context has been previously created for the contexts passed in the array,
+       and the shared-context is the same for all, then that one is returned.
+    -  |
+       If none of the contexts in the array belong to a shared context, then a shared context is created
+       encompassing all the contexts in the array.
+    -  |
+       If some of the contexts in the array are not yet associated with any shared context, and the rest
+       belong to a single shared context, then all are added to the same shared context.
+    -  |
+       If some of the contexts in the array already belong to different shared contexts,
+       then nullptr is returned.
+    -  |
+       If only one context is passed, then its shared context is returned if it already exists.
+       If not, then a new shared context is created and returned. Additional contexts
+       can then later be added to this one using $xDriverAddContextToSharedExp.
+    - "If only one context is passed, and that context is a shared context, then the same is returned."
```

Comment:
Does this API allow passing shared context as input ?

Reply:
@MichalMrozek : I dont see why not. In that case, the same shared-context would be returned. However, I see there could be a confusion if the interface is called with a mix of shared contexts and regular contexts, in that case, what the behavior should be? we could disallow that case but indicating that if the API is called with 1 or more shared-contexts, then nullptr is returned, which means, we dont create a shared-context of multiple shared-contexts.